[BE] feat: 관리자 페이지 카테고리&키워드 수정 기능 추가 #224

yuyoungrhee opened the pull request:
<!-- PR 제목 예시
ex. [BE] 로그인 API 구현
ex. [AN] 페이지 구현 -->

## 📌 변경 내용 & 이유
<!-- ex. 
- 사용자 로그인 기능 구현
- 로그인 실패 시 에러 메시지 반환 처리 추가 -->
- 관리자 키워드 관리 페이지에서 키워드의 `name` 수정 기능 추가
- 관리자 카테고리 관리 페이지에서 카테고리의 `name`, `colorCode` 수정 기능 추가

## 🧪 테스트 방법 (선택)
<!-- 코드 리뷰하는 팀원이 돌려볼 수 있도록 작성해주기 
ex. ReservationTest의 XXXX 테스트메서드로 확인 
ex. 어떤화면의 어떤기능에서 확인 -->

AdminKeywordController
    `@DisplayName("키워드를 수정할 수 있다.")`

AdminCategoryControllerTest
   `@DisplayName("카테고리를 수정할 수 있다.")`

## 🧩 관련 이슈
<!-- 이슈 태그: #123 -->
<!-- 이슈 닫기: close #123 -->
close #224 

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit


* **신규 기능**
  * 관리자 카테고리 및 키워드 목록에서 인라인 수정(이름/색상) 기능이 추가되었습니다.
  * 카테고리와 키워드 수정 API가 추가되었습니다.

* **버그 수정**
  * 카테고리 색상 입력값이 8자리에서 6자리(hex)로 변경되었습니다.

* **문서**
  * 관리자 홈 메뉴에 '수정' 기능이 반영되었습니다.
  * 히어릿 추가 관련 UI 문구가 '업로드'에서 '추가'로 변경되었습니다.
  * 관리자 내비게이션 및 메뉴 텍스트가 일부 변경되었습니다.

* **스타일**
  * 카테고리 및 키워드 관리 화면에 수정 버튼과 입력 필드가 추가되었습니다.

* **테스트**
  * 카테고리 및 키워드 수정 기능에 대한 테스트가 추가되었습니다.
  * 테스트 데이터의 색상 코드가 6자리 hex 형식으로 업데이트되었습니다.

* **기타**
  * 생성 및 수정 요청에 대한 입력값 유효성 검증이 강화되었습니다.
  * 카테고리 및 키워드 도메인에 이름 및 색상 코드 유효성 검증 로직이 추가되었습니다.
  * 카카오 로그인 관련 메서드명이 로그인과 회원가입을 통합하는 의미로 변경되었습니다.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

yuyoungrhee committed:
[BE] feat: 카테고리 수정 API 추가 #224

yuyoungrhee committed:
[BE] feat: 키워드 수정 API 추가 #224

yuyoungrhee committed:
[BE] feat: 관리자 페이지에서 수정 버튼 및 API 연결 #224

yuyoungrhee committed:
[BE] Merge branch 'develop-be' into feat/#224-admin-keyword-category-update #224

yuyoungrhee committed:
[BE] feat: requestDto 값 검증 추가 - null, blank 등 #224

yuyoungrhee committed:
[BE] fix: 히어릿 업로드 pathUrl 오류 수정 #224

yuyoungrhee committed:
[BE] refactor: SQL Query에서 limit가 아닌 size로 파라미터 의미 명확화 #224

Byesol reviewed (dismissed):
LGTM! 수고하셨어요!

gabean13 commented on backend/src/main/resources/templates/admin/home.html:
```diff
@@ -23,10 +23,10 @@ <h1>hEARit 관리자 페이지</h1>
                 </a>
                 <a th:href="@{/admin/categories}" class="list-group-item list-group-item-action">
-                    카테고리 관리 (조회, 추가)
+                    카테고리 관리 (조회, 추가, 수정)
```
카테고리랑은 관련없는데, 이 파일에서 히어릿 관리 (조회, 추가, 수정) 에서 히어릿 추가 삭제해야할 것 같아요!! 그리고 히어릿 업로드 말고 히어릿 추가 어떤가용 다른 메뉴들이랑 말을 통일하면 좋을 것 같아요

yuyoungrhee replied:
추가로 통일할게요!
[[BE] refactor: 관리자 페이지에서 히어릿 추가/업로드 중 추가로 용어 통일](https://github.com/woowacourse-teams/2025-hEARit/pull/228/commits/2dbceed217d889fdd5fa75d97c344acf021c2c77) https://github.com/woowacourse-teams/2025-hEARit/issues/224

gabean13 commented on backend/src/main/java/com/onair/hearit/domain/Category.java:
```diff
@@ -30,4 +30,9 @@ public Category(String name, String colorCode) {
         this.colorCode = colorCode;
     }
+
+    public void update(String name, String colorCode) {
+        this.name = name;
+        this.colorCode = colorCode;
+    }
 }
```
view에서 바로 내려오는 값인데 추가적인 검증이 필요할 것 같아요!!

yuyoungrhee replied:
인정합니다요
[[BE] feat: 카테고리, 키워드 도메인 검증 추가](https://github.com/woowacourse-teams/2025-hEARit/pull/228/commits/1254d0b4fe2b0e6da979e642b314366b668de8e2) https://github.com/woowacourse-teams/2025-hEARit/issues/224

gabean13 commented on backend/src/main/java/com/onair/hearit/admin/application/AdminKeywordService.java:
```diff
@@ -38,4 +41,12 @@ public void addKeyword(KeywordCreateRequest request) {
         Keyword keyword = new Keyword(request.name());
         keywordRepository.save(keyword);
     }
+
+
+    @Transactional
+    public void updateKeyword(Long keywordId, KeywordUpdateRequest request) {
```
여기 엔터 두개 들어갔어요

yuyoungrhee replied:
[[BE] refactor: 불필요한 개행 제거 및 컨벤션 맞게 수정](https://github.com/woowacourse-teams/2025-hEARit/pull/228/commits/5a54dc7b9c1ef2a04a528c5f9787027ca305da96) https://github.com/woowacourse-teams/2025-hEARit/issues/224

gabean13 commented on backend/src/main/java/com/onair/hearit/domain/Keyword.java:
```diff
@@ -26,4 +26,8 @@ public class Keyword {
         this.name = name;
     }
+
+    public void updateName(String newName) {
+        this.name = newName;
+    }
 }
```
여기도 동일하게 검증해주시는거 어떤가용

yuyoungrhee replied:
[[BE] feat: 카테고리, 키워드 도메인 검증 추가](https://github.com/woowacourse-teams/2025-hEARit/pull/228/commits/1254d0b4fe2b0e6da979e642b314366b668de8e2) https://github.com/woowacourse-teams/2025-hEARit/issues/224

Byesol commented on backend/src/main/java/com/onair/hearit/admin/presentation/AdminKeywordController.java:
```diff
@@ -41,8 +45,18 @@ public ResponseEntity<List<KeywordInfoResponse>> readAllKeywords() {
         return ResponseEntity.status(HttpStatus.CREATED).build();
     }
+
+
```
어 여기 엔터 두개에요

yuyoungrhee replied:
[[BE] refactor: 불필요한 개행 제거 및 컨벤션 맞게 수정](https://github.com/woowacourse-teams/2025-hEARit/pull/228/commits/5a54dc7b9c1ef2a04a528c5f9787027ca305da96) https://github.com/woowacourse-teams/2025-hEARit/issues/224

JO-eusan commented on backend/src/main/java/com/onair/hearit/admin/application/FileStorageService.java:
```diff
@@ -38,7 +38,7 @@ private String uploadToS3(MultipartFile multipartFile, String key) {
                     RequestBody.fromInputStream(inputStream, multipartFile.getSize())
             );
-            return key;
+            return "/" + key;
```
이게 빠졌었군요...

JO-eusan commented on backend/src/main/java/com/onair/hearit/admin/presentation/AdminCategoryController.java:
```diff
@@ -41,8 +45,17 @@ public ResponseEntity<List<CategoryInfoResponse>> readAllCategories() {
+    public ResponseEntity<Void> updateCategory(
+            @PathVariable Long categoryId,
+            @RequestBody @Valid CategoryUpdateRequest request
+    ) {
```
다른 메서드와 마찬가지로 파라미터 옆에 괄호를 붙이는거 어떨까요?

yuyoungrhee replied:
여기 왜이랬지 누구야 누가왔다갔어 귀신인가봐요 고쳤습니다.
[[BE] refactor: 불필요한 개행 제거 및 컨벤션 맞게 수정](https://github.com/woowacourse-teams/2025-hEARit/pull/228/commits/5a54dc7b9c1ef2a04a528c5f9787027ca305da96) https://github.com/woowacourse-teams/2025-hEARit/issues/224

JO-eusan commented on backend/src/main/java/com/onair/hearit/admin/presentation/AdminKeywordController.java:
```diff
@@ -41,8 +45,18 @@ public ResponseEntity<List<KeywordInfoResponse>> readAllKeywords() {
+    public ResponseEntity<Void> updateKeyword(
+            @PathVariable Long keywordId,
+            @RequestBody @Valid KeywordUpdateRequest request
+    ) {
```
여기도 마찬가지로 괄호를 파라미터 옆으로 옮기는거 어떨까요?

yuyoungrhee replied:
[[BE] refactor: 불필요한 개행 제거 및 컨벤션 맞게 수정](https://github.com/woowacourse-teams/2025-hEARit/pull/228/commits/5a54dc7b9c1ef2a04a528c5f9787027ca305da96) https://github.com/woowacourse-teams/2025-hEARit/issues/224

JO-eusan reviewed (changes requested):
관리자 페이지 수정 기능이랑 `/` 없는 오류 잘 수정해주셨네용.
다른 분들 커멘트 반영이 필요한 것 같아서 request changes로 커멘트 남깁니다.

yuyoungrhee committed:
[BE] refactor: 관리자 페이지에서 히어릿 추가/업로드 중 추가로 용어 통일 #224

yuyoungrhee committed:
[BE] refactor: 불필요한 개행 제거 및 컨벤션 맞게 수정 #224

yuyoungrhee committed:
[BE] feat: 카테고리, 키워드 도메인 검증 추가 #224

yuyoungrhee committed:
[BE] refactor: 카카오로그인 관련 메서드명 명확하게 개선 #224

yuyoungrhee committed:
[BE] fix: 잘못된 파라미터 검증 버그 수정 #224

yuyoungrhee commented on backend/src/main/java/com/onair/hearit/domain/Keyword.java:
```diff
@@ -24,6 +26,18 @@
+    public void updateName(String newName) {
+        validate(name);
+        this.name = newName;
+    }
```
[[BE] fix: 잘못된 파라미터 검증 버그 수정](https://github.com/woowacourse-teams/2025-hEARit/pull/228/commits/ca2fc30390f5544c6f4440c186806c88625251e7) https://github.com/woowacourse-teams/2025-hEARit/issues/224

오우예 감사합니다

yuyoungrhee committed:
[BE] feat: 카테고리&키워드 도메인 빈문자열 검증 추가 #224

yuyoungrhee commented on backend/src/main/java/com/onair/hearit/domain/Keyword.java:
```diff
@@ -24,6 +26,18 @@ public class Keyword {
+        if (name == null || name.length() > KEYWORD_NAME_MAX_LENGTH) {
+            throw new InvalidInputException("키워드 이름은 20자 이하여야 합니다.");
+        }
+    }
```
[[BE] feat: 카테고리&키워드 도메인 빈문자열 검증 추가](https://github.com/woowacourse-teams/2025-hEARit/pull/228/commits/12086aa7deb24daec8fee25a2424259c67fa594f) https://github.com/woowacourse-teams/2025-hEARit/issues/224

했어욘

gabean13 reviewed (dismissed):
고생하셔썽요 멍구

Byesol reviewed (dismissed):
수고하셨어요, 멍구

yuyoungrhee committed:
[BE] Merge branch 'develop-be' into feat/#224-admin-keyword-category-update #224

# Conflicts:
#	backend/src/main/java/com/onair/hearit/auth/application/AuthService.java
#	backend/src/main/java/com/onair/hearit/auth/presentation/AuthController.java
#	backend/src/test/java/com/onair/hearit/auth/application/AuthKakaoServiceTest.java

Byesol reviewed (approved):
merge conflict 해결에 따른 approve 할께염

JO-eusan reviewed (approved):
리뷰 확인했습니다. Approve & Merge 할게요!
